core/iwasm/common/iwasm_common.cmake

```diff
@@ -24,6 +35,12 @@ if (WAMR_BUILD_INVOKE_NATIVE_GENERAL EQUAL 1)
   #   in arm and mips need to be 8-bytes aligned, and some arguments
   #   of x86_64 are passed by registers but not stack
   set (source_all ${c_source_all} ${IWASM_COMMON_DIR}/arch/invokeNative_general.c)
+elseif (OSX_UNIVERSAL_BUILD)
+  if (NOT WAMR_BUILD_SIMD EQUAL 1)
+    set (source_all ${c_source_all} ${IWASM_COMMON_DIR}/arch/invokeNative_aarch64.s ${IWASM_COMMON_DIR}/arch/invokeNative_em64.s)
```

Comment:
Please see my reply above, but basically we need to include the both architectures in the source list to avoid linker and compiler errors. I realise it's a bit ugly to avoid using `WAMR_BUILD_TARGET` but I don't think there's a way around it unless we change `WAMR_BUILD_TARGET` to be a list of architectures to be compiled for e.g. `AARCH64;X86_64` for universal builds. [`CMAKE_OSX_ARCHITECTURES`](https://cmake.org/cmake/help/latest/variable/CMAKE_OSX_ARCHITECTURES.html) is a predefined cmake flag that is used to control which architectures are to be compiled for on osx.

Reply:
Will this shadow the below configurations or impact the original configurations? For example, in normal Darwin x86-64 building, it uses the below config `set (source_all ${c_source_all} ${IWASM_COMMON_DIR}/arch/invokeNative_em64.s)`, but after this PR, it is overwritten with the config you added.

Reply:
So this will only shadow below configurations if `CMAKE_OSX_ARCHITECTURES` contains entries for both arm and x86 i.e. a universal build. In this case, `WAMR_BUILD_TARGET` would be ignored. But this makes sense since if `CMAKE_OSX_ARCHITECTURES` is set to `arm;x86_64`, falling into one of the `WAMR_BUILD_TARGET` configuration below will always cause a compilation error (for the reasons discussed in the other thread). I hope that makes sense?


N.B. it only makes sense to set `CMAKE_OSX_ARCHITECTURES` on osx machines.

Reply:
OK, sounds good, I just hoped that it doesn't change the old configs, e.g. build iwasm on Linux x86-64, or on Darwin x86-64, or on Darwin aarch64, etc.